File: Java/1768.java
class Solution {
    public String mergeAlternately(String word1, String word2) {
        int min = Math.min(word1.length(), word2.length());
        StringBuilder finalWord = new StringBuilder();
        int i = 0;
        while (i < min) {
            finalWord.append(word1.charAt(i));
            finalWord.append(word2.charAt(i));
            i++;
        }
        if (word1.length() > i) finalWord.append(word1.substring(i));
        if (word2.length() > i) finalWord.append(word2.substring(i));
        return finalWord.toString();
    }
}

File: TS/1768.ts
function mergeAlternately(word1: string, word2: string): string {
    let stringFinal = "";
    const minLength = Math.min(word1.length, word2.length);
    let i = 0;
    for (i = 0; i < minLength; i++) {
        stringFinal += word1[i];
        stringFinal += word2[i];
    }
    if (word1.length > minLength) stringFinal += word1.substring(i);
    if (word2.length > minLength) stringFinal += word2.substring(i);
    return stringFinal;
};